### ISSUE TITLE
vscode.Diagnostic.relatedInformation is stale (incorrectly cached?) after it's updated with the relatedInformation removed

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode#151385]
Does this issue occur when all extensions are disabled?: Yes (but my repro is with the C/C++ extension)

- VS Code Version: 1.67.2 or
- Version: 1.68.0-insider (user setup)
Commit: 87630a9122f88f5062fe9964839a92116c274a86
Date: 2022-06-06T17:09:26.575Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.22000
- OS Version: Windows (I suspect it repros with all OS's)

Steps to Reproduce:

1. Do something to cause a vscode.Diagnostic object to be created that has a relatedInformation set.
2. Do something to cause the same diagnostic to be set but with the relatedInformation not set (either missing or set to undefined).

Bug: The VS Code UI doesn't remove the relatedInformation from the diagnostic in the Problems windows. If the Problems tab is switched to another tab and then bac, the child disappears, but the parent UI node still renders with an empty child expansion widget on the left. See the screenshot (top half is before a refresh is triggered, and the bottom half is after the UI refreshes):
![image](https://user-images.githubusercontent.com/19859882/172277608-16749f7e-69ea-411a-b3b1-f009135a4e3c.png)

Repro details for the C/C++ extension are:
1. Install the C/C++ version 1.10.4 (pre-release) or later (1.9.8 is not sufficient).
2. Add setting
```json
"C_Cpp.codeAnalysis.clangTidy.checks.enabled": [
        "*"
    ]
```
2. Create a test.cpp with
```cpp

#include <algorithm>
#include <cstdlib>
#include <numeric>
#include <atomic>
#include <cassert>
#include <cctype>
#include <cerrno>
#include <cfenv>
#include <cfloat>
#include <cinttypes>
#include <ciso646>
#include <climits>
#include <clocale>
#include <cmath>
#include <csetjmp>
#include <csignal>
#include <cstdarg>
#include <cstddef>
#include <cstdint>
#include <cstdio>
#include <cstdlib>
```
3. Use the "C/C++: Run Code Analysis on Active File" command.
4. Select the "Fix..." code action for the "duplicate include" problem. See the screenshot:
![image](https://user-images.githubusercontent.com/19859882/172279013-b354e98b-701d-442a-b236-710a393d8466.png)
5. Save the document and then wait for the flame icon in the status bar to disappear, which indicates the Problem window has been updated with a diagnostic that has the relatedInformation removed -- the correctness of this code has been verified via debugging our TypeScript code, and the fact that the child object disappears after the UI is "refreshed" (but with an empty child) also reinforces the conclusion that it's a VS Code UI issue, because our extension code is not involved at that point.

**Comment by sandy081** [comment]
This seems to be fixed. Please retry with latest VS Code and confirm. If you are still seeing the issue please provide the new steps to reproduce.

**Comment by sean-mcmanus** [comment]
Yeah, the related info is removed now. The UI is still incorrect and shows a collapse icon on the left even though there are no more children, but that is low priority.

![Image](https://github.com/user-attachments/assets/6d30ceca-4e30-48e9-a208-cb7b37dd8eba)
